x-pack/plugins/observability_solution/observability_ai_assistant_app/server/rule_connector/index.ts

```diff
@@ -172,9 +177,27 @@ async function executor(
     });
   });
 
-  const backgroundInstruction = getBackgroundProcessInstruction(
+  const backgroundInstruction = await getBackgroundProcessInstruction(
```

Comment:
I don't think you want to inject this as an `instruction`. Instructions are meant to augment the system message, which controls the behaviour of the LLM when answering a user's prompt. But it is lost for instance when the user opens the conversation. What you probably want here is to append a function request and response, similar to what you did for `get_connectors`. That way that context is always available in the conversation. The instruction should mostly be about the fact that it is running in a background process and the user cannot immediately provide clarification.

Reply:
Fixed in https://github.com/elastic/kibana/pull/181501/commits/bcd3b2b3de69478c46e9c5d1d4c390b074a4b587